oomph/c_output.py

```diff
@@ -250,9 +251,18 @@ def emit_funcdef(
             self.after_body += "return retval;"
 
         varnames = [self.variable_names[var] for var in funcdef.argvars]
+
+        if not static:
+            self.file_emitter.h_code += (
+                self.file_emitter.declare_function(c_name, functype, static=False)
+                + ";\n"
+            )
+
+        declaration = self.file_emitter.declare_function(
+            c_name, functype, varnames, static
+        )
         return f"""
-        {self.file_emitter.declare_function(c_name, functype, varnames, static)}
-        {{
+        {declaration} {{
```

Comment:
This doesn't need to be changed, can put `declare_function(...)` inside f string